### PR TITLE
Fix Aravis Buffer on Invalid Image from Camera.

### PR DESCRIPTION
**Pull request by wongosaurus**
Please check if this breaks your robot. We had issues where cameras stopped sending images after some timeouts, after close investigation, the camera was running out of buffers to insert images.

### CODE REVIEW

File: module/input/Camera/src/AravisCamera.cpp [hunk 1]
```diff
@@ -173,6 +173,14 @@ namespace input {
             context->reactor.emit<Scope::DIRECT>(msg);
             arv_stream_push_buffer(stream, buffer);
         }
+        else {
+            // In the case where it fails, reinsert the Buffer back into the stream, so we don't run out.
+            if (buffer != NULL) {
+                // std::cout << "Buffer Failed: " << context->cameraID << " " << arv_buffer_get_status(buffer) <<
```
Comment (TrentHouliston) [inline module/input/Camera/src/AravisCamera.cpp]: Don't commit commented code!

Comment (wongosaurus) [reply module/input/Camera/src/AravisCamera.cpp]: Done

File: module/input/Camera/src/AravisCamera.cpp [hunk 2]
```diff
@@ -173,6 +173,12 @@ namespace input {
             context->reactor.emit<Scope::DIRECT>(msg);
             arv_stream_push_buffer(stream, buffer);
         }
+        else {
```
Comment (Bidski) [inline module/input/Camera/src/AravisCamera.cpp]: I think I would prefer to this to be
```
if (buffer != NULL) {
    if (arv_buffer_get_status(buffer) == ARV_BUFFER_STATUS_SUCCESS) {
        // send message
        arv_stream_push_buffer(....); 
    }
    else {
        arv_stream_push_buffer(....);
    }
}
```

Whats your opinion @TrentHouliston ?

Comment (TrentHouliston) [reply module/input/Camera/src/AravisCamera.cpp]: I agree it should be changed to that, but I also think the compiler would probably optimise it to that anyway :P

Comment (Bidski) [reply module/input/Camera/src/AravisCamera.cpp]: readability concerns are more important than potential compiler optimisations?

Comment (wongosaurus) [reply module/input/Camera/src/AravisCamera.cpp]: Changes made, it matches the same format as the examples given in the documentation:
https://aravisproject.github.io/docs/aravis-0.4/ArvCamera.html

Comment (TrentHouliston) [reply module/input/Camera/src/AravisCamera.cpp]: > readability concerns are more important than potential compiler optimisations?

I said I agree!

**Pull request merged**